### PR TITLE
Blog post on understanding Entropy

### PR DESCRIPTION
**Pull request by sashwathn**
Please review and approve.

### CODE REVIEW

**Comment by andrewm4894** [comment]
update the date maybe

File: blog/2023-05-02-understanding-entropy-the-key-to-secure-cryptography-and-randomness/index.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+title: "Understanding Entropy: The Key to Secure Cryptography and Randomness"
+authors: satya
+tags: [entropy, security, cryptography, monitoring]
+keywords: [entropy, security, cryptography, monitoring]
+image: ./img/stacked-netdata.png
+---
+
+![stacked-netdata](./img/stacked-netdata.png)
+
+Entropy is a measure of the randomness or unpredictability of data. In the context of cryptography, entropy is used to generate random numbers or keys that are essential for secure communication and encryption. Without a good source of entropy, cryptographic protocols can become vulnerable to attacks that exploit the predictability of the generated keys.
```
Comment (andrewm4894) [inline blog/2023-05-02-understanding-entropy-the-key-to-secure-cryptography-and-randomness/index.md:12]:
```suggestion
[Entropy](https://en.wikipedia.org/wiki/Entropy_(computing)) is a measure of the randomness or unpredictability of data. In the context of cryptography, entropy is used to generate random numbers or keys that are essential for secure communication and encryption. Without a good source of entropy, cryptographic protocols can become vulnerable to attacks that exploit the predictability of the generated keys.
```

Comment (sashwathn) [reply blog/2023-05-02-understanding-entropy-the-key-to-secure-cryptography-and-randomness/index.md]: Done. From local repository.

**Comment by sashwathn** [comment]
@andrewm4894 : Approve once you check.

File: blog/2023-05-03-understanding-entropy-the-key-to-secure-cryptography-and-randomness/index.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+## Entropy in Cloud VMs
+
+Cloud-provided VMs typically have access to hardware-generated entropy, but the specific implementation depends on the cloud provider and the underlying infrastructure.
+
+Most cloud providers recognize the importance of hardware-generated entropy for their customers' cryptographic operations and security needs. They often utilize hardware random number generators (HRNGs) or hardware security modules (HSMs) in their infrastructure to provide a reliable source of entropy.
+
+In general, cloud providers expose this entropy to VMs through virtual devices or interfaces. This allows VMs to access the hardware-generated entropy as if it were a local resource.
+
+- **AWS (Amazon Web Services)**
```
Comment (cakrit) [inline blog/2023-05-03-understanding-entropy-the-key-to-secure-cryptography-and-randomness/index.md:100]: Same issue here as in the uptime blog, this list doesn't appear properly in HTML. Adding the ":" is the faster solution.

File: blog/2023-05-03-understanding-entropy-the-key-to-secure-cryptography-and-randomness/index.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+Similarly, in SSH, the client and server generate random numbers to establish a shared secret key for encryption and authentication. If the random numbers are predictable or biased, an attacker can potentially intercept the communication and compromise its security.
+
+To prevent this type of attack, SSH requires a sufficient amount of high-quality entropy to generate the random numbers. The client and server generate the random numbers by combining random data from various sources, such as hardware interrupts, keyboard presses, and mouse movements. This ensures that the random numbers are unpredictable and random.
+
+### Database servers
+
+Database servers can use entropy in various ways, depending on the specific database system and its configuration. Here are some examples:
+
+- **Encryption:** Many database systems support encryption of data-at-rest or data-in-transit, which relies on random numbers for generating encryption keys, initialization vectors, and other cryptographic elements. For example, Microsoft SQL Server provides Transparent Data Encryption (TDE), which encrypts the data stored on disk, and uses a random number generator to create the encryption key.
```
Comment (cakrit) [inline blog/2023-05-03-understanding-entropy-the-key-to-secure-cryptography-and-randomness/index.md:68]: I thought we're adding the bold before the ":". Minor, as I didn't notice it at all in the HTML anyway.

**Review by cakrit — CHANGES_REQUESTED** [review]

File: blog/2023-05-03-understanding-entropy-the-key-to-secure-cryptography-and-randomness/index.md (hunk 2 above)
Comment (sashwathn) [reply blog/2023-05-03-understanding-entropy-the-key-to-secure-cryptography-and-randomness/index.md]: Fixed.

**Comment by sashwathn** [comment]
@cakrit : All changes are made. Please approve.

**Pull request merged**